apps/designer/app/designer/features/style-panel/shared/get-final-value.ts

```diff
@@ -17,5 +24,15 @@ export const getFinalValue = ({
   if (currentValue?.value === "inherit" && inheritedValue !== undefined) {
     return inheritedValue;
   }
+  if (
+    currentValue &&
+    currentValue.type === "unit" &&
+    isValid(
+      property,
+      toValue({ ...currentValue, value: Math.abs(currentValue.value) })
+    ) === false
+  ) {
+    return { ...(currentValue as unknown as InvalidValue), type: "invalid" };
```

Comment:
same here with assertion

Reply:
The reported error here is just terrible:
> Type '"invalid"' is not assignable to type '"unit"'.ts(2322)

The presence of the error is legit though and by casting here we risk some runtime failures elsewhere. You want to end up with `InvalidValue` here. That type has its `value` property typed as `string`:
https://github.com/webstudio-is/webstudio-designer/blob/cfd40d748b936d23caa3b6c60a8385c5df7f77c8/packages/react-sdk/src/css/schema.ts#L34
but the `currentValue`'s type here is `UnitValue` (we are within `if (currentValue.type === "unit")` here) and its `value` property is typed as `number`:
https://github.com/webstudio-is/webstudio-designer/blob/cfd40d748b936d23caa3b6c60a8385c5df7f77c8/packages/react-sdk/src/css/schema.ts#L12